optaplanner-examples/src/main/java/org/optaplanner/examples/vehiclerouting/optional/score/VehicleRoutingIncrementalScoreCalculator.java

```diff
@@ -1,223 +0,0 @@
-package org.optaplanner.examples.vehiclerouting.optional.score;
-
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-
-import org.optaplanner.core.api.score.buildin.hardsoftlong.HardSoftLongScore;
-import org.optaplanner.core.api.score.calculator.IncrementalScoreCalculator;
-import org.optaplanner.examples.vehiclerouting.domain.Customer;
-import org.optaplanner.examples.vehiclerouting.domain.Standstill;
-import org.optaplanner.examples.vehiclerouting.domain.Vehicle;
-import org.optaplanner.examples.vehiclerouting.domain.VehicleRoutingSolution;
-import org.optaplanner.examples.vehiclerouting.domain.timewindowed.TimeWindowedCustomer;
-import org.optaplanner.examples.vehiclerouting.domain.timewindowed.TimeWindowedVehicleRoutingSolution;
-
-public class VehicleRoutingIncrementalScoreCalculator
```

Comment:
This is an unfortunate loss.
We were making very good use of this calculator during the recent competitions.

Reply:
There are two options.

Option 1: Ignore the list variable change notifications and calculate the score only based on vehicle, previous, and next shadow variables. If the incremental score director stops throwing UnsupportedOEs and just does nothing when used with a list variable, Option A can be implemented now because it can be done without adding the list variable related before/after methods to the public `IncrementalScoreCalculator` interface.

Downside: I tried this but gave up because it seemed extremely difficult to correctly process _individual_ shadow variable changes. The calculator is not being triggered after all shadows are up-to-date. It's being triggered _immediately_ before and after the shadow changes. So it's constantly working on an inconsistent state of the shadow entities.

Option 2: Delete the calculator now and add it back when the list variable APIs are stable and made public. I tried this to get an early confirmation that the proposed `ListVariableListener` methods for subList changes are compatible with incremental score calculation. I re-implemented the calculator successfully and it seems even faster than the chained one.